### ISSUE TITLE
How to create Cmd.bind

### ISSUE THREAD

**Issue by Evangelink** [elmish/Elmish.WPF#380]
I need to postpone the execution of the cmdMsgs to after the point where one of my UI element is being initialized/loaded (I am changing some state at this point). Currently, my `cmdMsgsToCmd` function looks like:

```fsharp
let cmdMsgsToCmd = function
	| CmdMsg1 -> ...
	| CmdMsg2 -> ...
```

and I wanted to convert it into something like:

```fsharp
let cmdMsgsToCmd cmdMsg =
	Cmd.ofAsync.result (Dispatch.onThreadPool (async {
		while not automation.Value.Active do
            do! Async.Sleep 50
	}))
    |> Cmd.bind (fun () -> 
		match cmdMsg with
		| CmdMsg1 -> ...
		| CmdMsg2 -> ...)
```

but the `Cmd.bind` function does not exist and I am not sure how to create it.

Maybe I could do things differently but the main thing is that I want to have the various cmdMsg to "wait" (but not block UI thread) until the initialization of the UI object. Once this is done all objects will behave normally.

**Comment by TysonMN** [comment]
1. `Cmd.bind` is an interesting idea.  My intuition it that it could exist, but I know it doesn't right now.  I will look into this.

 2. Try solving your problem using `Cmd.ofSub`. It fully exposes what a `Cmd` is, so if it can't be done with that, the it can't be done at all.

3. You should be able to slove your problem via pushing instead of polling.  If you provide a reproduction, then I can implement that for you.

**Comment by Evangelink** [comment]
I have tried to implement it by having the `Loaded` event to push a message to the update loop but this doesn't prevent all cmdMsg to be executed (doesn't delay them). Best option would still be to have a `Cmd.bind`.

For now, I have managed to "workaround" by having only the few UI related cmdMsg to call the wait for active automation.

I am a bit in a rush of finishing up an app for the release in a few weeks but I will try to have a reproducer by then.

Thanks again.

**Comment by TysonMN** [comment]
> 1. `Cmd.bind` is an interesting idea.  My intuition it that it could exist, but I know it doesn't right now.  I will look into this.

I tried but failed.  Now it seems to me like it is not possible.

**Comment by TysonMN** [comment]
> 2\. Try solving your problem using `Cmd.ofSub`. It fully exposes what a `Cmd` is, so if it can't be done with that, the it can't be done at all.

Try this.

```fsharp
type MyCmdMsg =
  | CmdMsg1
  | CmdMsg2

let isReady () = true // TODO: automation.Value.Active
let currentCmdMsgToCmd = function
  | CmdMsg1 -> () // TODO: replace
  | CmdMsg2 -> () // TODO: replace

let cmdMsgsToCmd (cmdMsg: MyCmdMsg) : Cmd<_> =
  let f dispatch =
    async {
      while not <| isReady () do
        do! Async.Sleep 50
        cmdMsg |> currentCmdMsgToCmd |> dispatch
    } |> Async.Start
  Cmd.ofSub f
```

**Comment by TysonMN** [comment]
Ah! Bug in the first version of that code, so copy it from GitHub and not from a notification or email.

**Comment by TysonMN** [comment]
Closed due to inactivity. Feel free to reopen.

**Comment by Evangelink** [comment]
I have replaced the 2 problematic parts in my code with a mailbox. The cmd msg now simply pushes a message in the mailbox and the mailbox is tuned to have the await and process messages in a synchronous way.
